Support React 16

jfsiii opened the pull request:
From [the React 16 announcement](https://reactjs.org/blog/2017/09/26/react-v16.0.html#packaging)

> The deprecations introduced in 15.x have been removed from the core package. `React.createClass` is now available as `create-react-class`, `React.PropTypes` as `prop-types` ...

  * Replaced `React.PropTypes` and `React.createClass` with their npm packages
  * Updated `react` and `react-dom` versions in `peerDependencies`

jfsiii committed:
Remove use of deprecated `React.PropTypes`

jfsiii committed:
Add prop-types dependency

Also update react & react-dom versions

jfsiii committed:
Merge pull request #1 from jfsiii/jfsiii-remove-react-proptypes

Remove use of deprecated `React.PropTypes`

jfsiii committed:
Replace React.createClass

Like `React.propTypes`, deprecated in 15.x & removed in 16

jfsiii committed:
Add create-react-class dependency

jfsiii committed:
Merge pull request #2 from jfsiii/jfsiii-replace-react-createclass

Replace React.createClass

0xjocke commented:
Was going to make a similar PR.

`create-react-class` and `prop-types` should probably be dependencies instead of peer-dependencies?

contra commented:
@bachstatter The PR has them in peer deps

jfsiii committed:
Move `react-create-class` & `prop-types` to dependencies

Can't rely on users having these installed and the code directly depends on them

jfsiii commented:
@bachstatter Good catch. I've updated the PR.

jfsiii commented:
@vicapow do you have any questions or concerns? Is there anything I can do to help get this merged?

yocontra commented:
Looks like I can merge, but I don't think I can publish to npm.

yocontra commented:
Ah, I do have publish access. Published as 1.1.4.
